### DISCUSSION TITLE
Packaging Ark and Positron for NixOS

### DISCUSSION THREAD

**Discussion by b-rodrigues** [posit-dev/positron#4542] [Q&A]
Hello

I've been trying to package Positron for NixOS. My first attempt was to use the deb package and patch to make it work for NixOS, which wasn't successful. So now I'm trying to build it from source. During the build process it turns out that it checks for Ark, and if it's not found then the user gets prompted to install it automatically.

This however can't work for NixOS, as builds happen in a sandbox without internet connection, to ensure reproducibility. So assets and dependencies need to be downloaded and built beforehand.

So now I've built Ark, [here's the expression](https://github.com/b-rodrigues/nixpkgs/blob/5abced36a16f47f25cf00d210da88be4b8123711/pkgs/by-name/po/posit-ark/package.nix) that builds it.

I got it working in a jupyter notebook session, and it seems to work, but I was wondering what else I should do to properly test the build (independently from positron). Is there a series of tests that I should/could run?

**Comment by jmcphers** [comment]
Nice work! As you probably discovered, you can just build Ark alongside Positron and it'll use that instead of trying to download from Github. 

There are a number of automated tests you can run on Positron. The best place to learn about these is in the Github workflows (in `.github/workflows`). Here's a few you can try on your build: 

https://github.com/posit-dev/positron/blob/08986c170f3cea9ddda6415387ae6d85b99cbcda/.github/workflows/ci.yml#L71-L95
